system/HTTP/CURLRequest.php

```diff
@@ -675,7 +678,13 @@ protected function setCURLOptions(array $curl_options = [], array $config = [])
 		if ($config['debug'])
 		{
 			$curl_options[CURLOPT_VERBOSE] = 1;
-			$curl_options[CURLOPT_STDERR]  = is_string($config['debug']) ? fopen($config['debug'], 'a+') : fopen('php://output', 'w+');
+
+			if ($config['debug'] === true)
+			{
+				$curl_options[CURLOPT_STDERR] = fopen('php://temp', 'w+');
```

Comment:
`fopen('php://output', 'w+')` is working just fine for outputting things. We should stay with that because otherwise it just complicates things later. So I would just go with the initial one-liner.

Reply:
For me this will produce error: CRITICAL - 2019-09-09 21:04:32 --> curl_setopt_array(): supplied argument is not a valid File-Handle resource

Reply:
We definitely want `php://output` as that directs tot he buffered output, whereas `php://temp` would require harvesting the contents from the temp directory.

The issue may be that `php://output` is explicitly write-only, and `w+` specifies opening a resource for "reading and writing" which may cause errors on some systems and not others (just a guess). Try using `fopen('php://output', 'w')`

Reply:
Actually reviewing the docs it states "this will enable additional debugging to echo to STDOUT during the script execution" so we may want to use `fopen('php://stdout', 'w')` which bypasses buffering altogether. Hardcore debugging :)

Reply:
@MGatner I tried: 
- fopen(php://output 'w') -> produce CRITICAL - 2019-09-24 12:07:11 --> curl_setopt_array(): cannot represent a stream of type Output as a STDIO FILE*
- fopen(php://stdout 'w') ->produce nothing (empty output on screen but maybe something in CLI)
- STDOUT ->produce CRITICAL - 2019-09-24 12:10:04 --> Use of undefined constant STDOUT-assumed 'STDOUT' (this will throw an Error in a future version of PHP)

Only working solution how to print to screen is from php://temp or any file that we can make but this is temporary and we do not have to delete it afterwards. I am not sure about output in CLI.

Reply:
Actually, I tested `fopen('php://stderr', 'w+')` - it works fine, plus it play nice with `php spark serve`.

@MGatner in the past, Windows used to have an old versions of libraries - which suffered from the bugs. I'm sure it have changed. In this case the difference may be the type of the server (apache, nginx) - and in what mode does php work? Idk... it's a blind shot.

Reply:
@michalsn yep but it will work only in CLI and not like the output to screen during the regular request (if it is wanted the behavior to debug only in CLI).

Reply:
@xbotkaj A quick internet search reveals that this definitely is an issue that crops up with Windows PHP cURL. One question (from https://github.com/guzzle/guzzle/issues/1413#issuecomment-222031665), are you testing against the same server? If so, can you test against some external site and see if you still get the error with `fopen('php://output', 'w')`? I believe that is our desired stream since `php://stdout` doesn't seem to write to the web-accessible buffer.

Reply:
@MGatner It is confirmed. It is a windows problem. I uploaded it to my linux server and it works without problem. :)